spec/support/saml_response_helper.rb

```diff
@@ -34,6 +34,13 @@ def raw_xml_response
       end
     end
 
+    def saml_response(settings)
+      @_saml_response ||= OneLogin::RubySaml::Response.new(
+        raw_xml_response,
+        settings: settings
+      )
+    end
```

Comment:
alas no. The tests for SamlIdpController depend on the original module method, as it is not an instance method on the `XmlDoc` class like the latter is.